roles/backup/tasks/postgres.yml

```diff
@@ -0,0 +1,89 @@
+---
+
+- name: Check for specified PostgreSQL configuration
+  k8s_info:
+    kind: Secret
+    namespace: '{{ meta.namespace }}'
+    name: '{{ tower_postgres_configuration_secret }}'
+  register: _custom_pg_config_resources
+  when: tower_postgres_configuration_secret | length
+
+- name: Check for default PostgreSQL configuration
+  k8s_info:
+    kind: Secret
+    namespace: '{{ meta.namespace }}'
+    name: '{{ tower_name }}-postgres-configuration'
+  register: _default_pg_config_resources
+
+- name: Set PostgreSQL configuration
+  set_fact:
+    pg_config: '{{ _custom_pg_config_resources["resources"] | default([]) | length | ternary(_custom_pg_config_resources, _default_pg_config_resources) }}'
+
+- name: Store Database Configuration
+  set_fact:
+    awx_postgres_user: "{{ pg_config['resources'][0]['data']['username'] | b64decode }}"
+    awx_postgres_pass: "{{ pg_config['resources'][0]['data']['password'] | b64decode }}"
+    awx_postgres_database: "{{ pg_config['resources'][0]['data']['database'] | b64decode }}"
+    awx_postgres_port: "{{ pg_config['resources'][0]['data']['port'] | b64decode }}"
+    awx_postgres_host: "{{ pg_config['resources'][0]['data']['host'] | b64decode }}"
+
+- name: Get the postgres pod information
+  k8s_info:
+    kind: Pod
+    namespace: '{{ meta.namespace }}'
+    label_selectors:
+      - "app={{ tower_name }}-{{ deployment_type }}-postgres"
+  register: postgres_pod
+  until: "postgres_pod['resources'][0]['status']['phase'] == 'Running'"
+  delay: 5
+  retries: 60
```

Comment:
This really only allows backup if the user makes use of the **managed** Postgres Statefulset. If the user supplies their own container-based database it would either need to be documented that they apply the above label or you could provide them an override option.

In the pulp-operator a label selector can be provided in the CRD. 
https://github.com/pulp/pulp-operator/blob/main/deploy/crds/pulpproject_v1beta1_pulp_crd.yaml#L100-L102
https://github.com/pulp/pulp-operator/blob/main/roles/postgres/tasks/migrate_data.yml#L11-L25

Just a suggestion, it may not be what you are looking to support.

Reply:
That makes sense.  I've added this to the data migration, backup, and restore tasks which look for a postgres pod. 

Commit: https://github.com/ansible/awx-operator/pull/133/commits/091c11eba91243a9180d8401f694fcaefac80d59